File: src/__tests__/ApexLexerTest.ts
import { ApexLexer } from "../ApexLexer";
import { CaseInsensitiveInputStream } from "../CaseInsensitiveInputStream"
import { ANTLRInputStream, CommonTokenStream } from 'antlr4ts';

test('Lexer generates tokens', () => {
    let lexer = new ApexLexer(new ANTLRInputStream("public class Hello {}"))
    let tokens  = new CommonTokenStream(lexer);
    expect(tokens.getNumberOfOnChannelTokens()).toBe(6)
})

test('Case insensitivity', () => {
    let lexer = new ApexLexer(new CaseInsensitiveInputStream("public"))
    let tokens  = new CommonTokenStream(lexer);
    expect(tokens.getNumberOfOnChannelTokens()).toBe(2)
})

test('Case insensitivity (upper case)', () => {
    let lexer = new ApexLexer(new CaseInsensitiveInputStream("PUBLIC"))
    let tokens  = new CommonTokenStream(lexer);
    expect(tokens.getNumberOfOnChannelTokens()).toBe(2)
})

test('Case insensitivity (mixed case)', () => {
    let lexer = new ApexLexer(new CaseInsensitiveInputStream("PuBliC"))
    let tokens  = new CommonTokenStream(lexer);
    expect(tokens.getNumberOfOnChannelTokens()).toBe(2)
})

File: jvm/src/test/java/com/nawforce/apexparser/ApexLexerTest.java
package com.nawforce.apexparser;

import org.antlr.v4.runtime.ANTLRInputStream;
import org.antlr.v4.runtime.CommonTokenStream;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.io.StringReader;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class ApexLexerTest {

    @Test
    void testLexerGeneratesTokens() {
        ApexLexer lexer = new ApexLexer(new ANTLRInputStream("public class Hello {}"));
        CommonTokenStream tokens  = new CommonTokenStream(lexer);
        assertEquals(6, tokens.getNumberOfOnChannelTokens());
    }

    @Test
    void testCaseInsensitivityLowerCase() throws IOException {
        ApexLexer lexer = new ApexLexer(new CaseInsensitiveInputStream(new StringReader("public")));
        CommonTokenStream tokens  = new CommonTokenStream(lexer);
        assertEquals(2, tokens.getNumberOfOnChannelTokens());
    }

    @Test
    void testCaseInsensitivityUpperCase() throws IOException {
        ApexLexer lexer = new ApexLexer(new CaseInsensitiveInputStream(new StringReader("PUBLIC")));
        CommonTokenStream tokens  = new CommonTokenStream(lexer);
        assertEquals(2, tokens.getNumberOfOnChannelTokens());
    }

    @Test
    void testCaseInsensitivityMixedCase() throws IOException {
        ApexLexer lexer = new ApexLexer(new CaseInsensitiveInputStream(new StringReader("PuBliC")));
        CommonTokenStream tokens  = new CommonTokenStream(lexer);
        assertEquals(2, tokens.getNumberOfOnChannelTokens());
    }
}
